numpy/distutils/ccompiler_opt.py

```diff
@@ -1462,34 +1470,39 @@ def feature_test(self, name, force_flags=None):
         if not os.path.exists(test_path):
             self.dist_fatal("feature test file is not exist", test_path)
 
-        test = self.dist_test(test_path, force_flags + self.cc_flags["werror"])
+        test = self.dist_test(
+            test_path, force_flags + self.cc_flags["werror"], macros
```

Comment:
Should `macros` be passed down to `dist_test` via `macros=macros`, for consistency?

Reply:
Yes, It would be better